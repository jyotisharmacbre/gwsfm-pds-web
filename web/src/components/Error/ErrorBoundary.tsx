import React from "react";
import { SeverityLevel } from "@microsoft/applicationinsights-web";
import TelemetryContext from "../../contexts/Telemetry/TelemetryContext";
import Error from '../../views/Error/Error';
import ErrorType from "../../enums/ErrorType";
import { History, Location } from 'history';

interface IProps {
    showPage?:boolean
    onError?: any;
    history: History;
    location: Location
}

class ErrorBoundary extends React.Component<IProps> {
    static contextType = TelemetryContext
    state = { hasError: false  };

    componentDidCatch(error, info) {
        const appInsights = this.context
        this.setState({ hasError: true });
        appInsights && appInsights.trackException({
            error: error,
            exception: error,
            severityLevel: SeverityLevel.Error,
            properties: { ...info }
        });
    }

    render() {
        if (this.state.hasError) {
            const { showPage, onError } = this.props;
            return (showPage != undefined && showPage) ? <Error history={this.props.history} location={this.props.location} type={ErrorType.warning}></Error> : onError == undefined ? <h1>Something went wrong</h1>
                : typeof onError === "function"
                ? onError()
                : React.createElement(onError);
        }

        return this.props.children;
    }
}

export default ErrorBoundary