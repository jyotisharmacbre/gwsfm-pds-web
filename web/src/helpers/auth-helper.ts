// @ts-ignore
import authentication from '@kdpw/msal-b2c-react';
import jwtDecode from 'jwt-decode';
import { DecodedToken } from '../models/decodedToken';

export const getDisplayName = () => {
  const decoded: DecodedToken = decodeToken();
  return decoded.name;
};

export const getFirstName = () => {
  const decoded: DecodedToken = decodeToken();
  return decoded.given_name;
};

export const getDisplayEmail = () => {
  const decoded: DecodedToken = decodeToken();
  return decoded.email ?
    decoded.email :
    decoded.emails && decoded.emails.length > 0 ?
      decoded.emails[0] :
      '';
};

export const logOut = () => {
  authentication.signOut();
}

function decodeToken() {
  const token = authentication.getAccessToken();
  const decoded: DecodedToken = jwtDecode(token);
  return decoded;
}

