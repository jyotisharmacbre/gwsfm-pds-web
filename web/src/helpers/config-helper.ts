const appConfig = () => {
    const config: any = (window as any).Config;
    if(config && config.REACT_APP_AUTH_INSTANCE==="#{AUTH_INSTANCE}#"){
      return {
        ...process.env
      }
    }
    return {
      ...config
    }
  };
  
  export default appConfig;