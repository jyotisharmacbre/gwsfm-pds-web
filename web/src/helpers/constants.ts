export const selectionButtons = [
  {
    title: 'BUTTON_YES',
    value: true
  },
  {
    title: 'BUTTON_NO',
    value: false
  }
];

export enum Notify {
  none,
  success,
  error
}
