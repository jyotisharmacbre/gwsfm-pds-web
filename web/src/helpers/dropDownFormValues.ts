export const projectStatusData: object[] = [
  { label: '1', value: '1' },
  { label: '2', value: '2' },
  { label: '3', value: '3' }
];

export const engagementData: object[] = [
  { label: 'Type 1', value: '1' },
  { label: 'Type 2', value: '2' },
  { label: 'Type 3', value: '3' }
];

export const typesOfEngagement: object[] = [
  {
    label: '---Please choose an option---',
    value: ''
  },
  {
    label: 'Fixed Price',
    value: 'Fixed Price'
  },
  {
    label: 'Time and Materials',
    value: 'Time and Materials'
  },
  {
    label: 'Risk and Reward',
    value: 'Risk and Reward'
  },
  {
    label: 'Cost Plus',
    value: 'Cost Plus'
  }
];

export const contractType: object[] = [
  {
    label: '---Please choose an option---',
    value: ''
  },
  {
    label: 'CT 1',
    value: 'CT 1'
  },
  {
    label: 'CT 2',
    value: 'CT 2'
  }
];

export const currency: object[] = [
  {
    label: '---Please choose an option---',
    value: ''
  },
  {
    label: 'AED (UAE Dirham)',
    value: 'AED'
  },
  {
    label: 'AFN (Afghani)',
    value: 'AFN'
  },
  {
    label: 'ALL (Lek)',
    value: 'ALL'
  },
  {
    label: 'AMD (Dram)',
    value: 'AMD'
  },
  {
    label: 'ANG (Dutch Antilles Guilder)',
    value: 'ANG'
  },
  {
    label: 'AOA (Kwanza)',
    value: 'AOA'
  },
  {
    label: 'ARS (Peso)',
    value: 'ARS'
  }
];

export const projectStatus: object[] = [
  {
    label: '---Please choose an option---',
    value: ''
  },
  {
    label: 'Initial Customer Inquiry',
    value: 'Initial Customer Inquiry'
  },
  {
    label: 'J&A',
    value: 'J&A'
  },
  {
    label: 'Archived',
    value: 'Archived'
  },
  {
    label: 'Lost',
    value: 'Lost'
  },
  {
    label: 'Order Received',
    value: 'Order Received'
  },
  {
    label: 'CPP',
    value: 'CPP'
  },
  {
    label: 'Complete',
    value: 'Complete'
  }
];

export const assetsWorkedOn: object[] = [
  {
    label: '---Please choose an option---',
    value: ''
  },
  {
    label: 'AS 1',
    value: 'AS 1'
  },
  {
    label: 'AS 2',
    value: 'AS 2'
  }
];

export const enquiryTypeData: object[] = [
  { label: 'Written', value: '1' },
  { label: 'Verbal', value: '2' },
  { label: 'Full tender enquiry with specifications & orgs', value: '3' },
  { label: 'Other', value: '4' }
];
