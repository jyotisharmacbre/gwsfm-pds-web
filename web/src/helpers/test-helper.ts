import checkPropTypes from 'check-prop-types';

export const findByTestAtrr = (component: any, attr: string) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

export const checkProps = (component: any, expectedProps: any) => {
  const propsErr = checkPropTypes(
    component.propTypes,
    expectedProps,
    'props',
    component.name
  );
  return propsErr;
};
