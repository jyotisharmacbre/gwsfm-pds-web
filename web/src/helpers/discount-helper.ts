import DiscountType from '../enums/DiscountType';
import { restrictMinusAndAllowDecimalForMaxRangeHundred, restrictMinusAndAllowDecimal } from '../helpers/utility-helper';

const normalizeClientDiscount = (discountType: DiscountType, discountValue) => {
    let updatedDiscount;
    if (discountType === DiscountType.Percent) {
        updatedDiscount = restrictMinusAndAllowDecimalForMaxRangeHundred(discountValue);
    }
    else updatedDiscount = restrictMinusAndAllowDecimal(discountValue);
    return updatedDiscount;
}

export default normalizeClientDiscount;