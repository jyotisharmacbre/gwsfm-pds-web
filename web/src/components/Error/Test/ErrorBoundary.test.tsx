import React from 'react';
import { mount } from 'enzyme';
import ErrorBoundary from '../ErrorBoundary';
import { IntlProvider } from 'react-intl';
import translations from '../../../Translations/translation';
import { findByTestAtrr } from '../../../helpers/test-helper';
import { MemoryRouter } from 'react-router-dom';

let wrapper;
let props = {
	showPage: false
}
const error = new Error('test');
const NullComponent = () => null;
const mountComponent = (props) => {
	wrapper = mount(
		<MemoryRouter>
		<IntlProvider locale="en" messages={translations['en'].messages}>
			<ErrorBoundary {...props}><NullComponent /></ErrorBoundary>
		</IntlProvider>
		</MemoryRouter>);
	};
describe('Error boundary test cases', () => {
	beforeEach(() => {
		mountComponent(props);
	});
	it('defines the component', () => {
		expect(wrapper).toBeDefined();
	});
	it('should match the snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('should not render error message when their is no error', () => {
		expect(findByTestAtrr(wrapper, 'error_message').exists()).toBeFalsy();
	})
	it('should render error message when any error occurred', () => {
		wrapper.find(NullComponent).simulateError(error);
		expect(findByTestAtrr(wrapper, 'error_message').exists()).toBeTruthy();
	})
	it('should render error page when any error occurred and showPage is set to true', () => {
		props.showPage = true;
		mountComponent(props);
		wrapper.find(NullComponent).simulateError(error);
		expect(findByTestAtrr(wrapper, 'error_page').exists()).toBeTruthy();
	})
});
