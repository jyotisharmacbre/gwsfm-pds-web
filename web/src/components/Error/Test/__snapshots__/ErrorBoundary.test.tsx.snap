// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error boundary test cases should match the snapshot 1`] = `ReactWrapper {}`;
